# 05-5 敘述-try


### 用try敘述捕捉不預期的例外.


#### (1)程式範例

```python
try:
    score=int(input('輸入成績'))

    if score>=60:
        print('及格')
    else:
        print('不及格')
except ValueError:
    print('輸入錯誤')
```


執行結果:
```
輸入成績 a
輸入錯誤
```


#### (2)程式範例

```python
try:
    score=int(input('輸入成績'))

    if score>100 or score<0:
        raise ValueError  #產生一個例外
    else:
        if score>=60:
            print('及格')
        else:
            print('不及格')
except ValueError:
    print('輸入錯誤')        
```


執行結果:
```
輸入成績 0
不及格
```
